src/state/auth/selectors.js

```diff
@@ -1,7 +1,22 @@
 export function getUserInfo(state, defaultsTo = null) {
-  return state.auth.loginInfo.get('userInfo', defaultsTo);
+  const got = state.auth.loginInfo.get('userInfo', defaultsTo);
+  if (got && typeof got === 'string') {
+    return JSON.parse(got);
+  }
+  return got;
 }
 
 export function getAccessToken(state, defaultsTo = null) {
-  return state.auth.loginInfo.get('accessToken', defaultsTo);
+  // XXX Terrible name
+  const got = state.auth.loginInfo.get('accessToken', defaultsTo);
+  if (got && typeof got === 'string') {
+    console.log('GOT:', got);
+    try {
+      return JSON.parse(got);
+    } catch (ex) {
+      console.warn('Unrecognized accessToken string:', got);
+      throw ex;
+    }
+  }
+  return got;
 }
```

Comment:
Please fix all of this. 

All I know is that it's a bad idea to call `accessToken` big string that is JSON that contains a key called `accessToken`.